File: backend/app/routers/catalog.py
"""Vehicle make/model catalog — source of truth for the frontend dropdowns.

Read-only, available to any authenticated user (the ticket-creation form is used
by Manager, Secretary, and Mechanic).
"""
from fastapi import APIRouter, Depends, Query
from sqlalchemy import distinct, select
from sqlalchemy.orm import Session

from ..database import get_db
from ..deps import ALL_ROLES, require_roles
from ..models import VehicleModel

router = APIRouter(prefix="/api/catalog", tags=["catalog"])


@router.get("/manufacturers")
def list_manufacturers(
    db: Session = Depends(get_db),
    _: dict = Depends(require_roles(*ALL_ROLES)),
):
    """All known manufacturers, sorted."""
    return list(
        db.scalars(
            select(distinct(VehicleModel.manufacturer)).order_by(VehicleModel.manufacturer)
        ).all()
    )


@router.get("/models")
def list_models(
    manufacturer: str = Query(...),
    db: Session = Depends(get_db),
    _: dict = Depends(require_roles(*ALL_ROLES)),
):
    """Models for a given manufacturer (for the cascading dropdown)."""
    return list(
        db.scalars(
            select(VehicleModel.model)
            .where(VehicleModel.manufacturer == manufacturer)
            .order_by(VehicleModel.model)
        ).all()
    )

File: frontend/src/services/catalog.ts
import apiClient from './apiClient';

// Vehicle make/model catalog (Stage backend follow-up). Read-only, available to
// any authenticated role. Powers the cascading manufacturer -> model dropdowns in
// the ticket and vehicle forms. The backend still accepts free-text
// manufacturer/model, so the UI keeps an "אחר" escape hatch for values not in the
// catalog; these services only provide the suggestion lists.

// GET /api/catalog/manufacturers — all known manufacturer names, sorted.
export async function getManufacturers(): Promise<string[]> {
  const { data } = await apiClient.get<string[]>('/catalog/manufacturers');
  return data;
}

// GET /api/catalog/models?manufacturer=... — models for the selected manufacturer.
export async function getModels(manufacturer: string): Promise<string[]> {
  const { data } = await apiClient.get<string[]>('/catalog/models', {
    params: { manufacturer },
  });
  return data;
}
